### PR TITLE
Bound ack time

### PR DESCRIPTION
**Pull request by idegtiarenko**
Currently, our default wait time for ack is unbound. This could result
in an entire test suite timeout rather than individual test cases failures.

We should also consider merging this to master.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java [hunk 1]
```diff
@@ -100,10 +103,6 @@ public static void assertAcked(AcknowledgedResponse response) {
         assertThat(response.getClass().getSimpleName() + " failed - not acked", response.isAcknowledged(), equalTo(true));
     }
 
-    public static void assertAcked(DeleteIndexRequestBuilder builder) {
-        assertAcked(builder.get());
-    }
```
Comment (idegtiarenko) [inline test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java:105]: This overloaded variant is covered by ones above and is actually not needed

**Comment by DaveCTurner** [comment]
Seems reasonable, or at least it's in line with our preference for failing tests more quickly than the suite timeout. I'm not sure 30s is a good value tho, there might be some tests that need a longer wait which this would cause to fail. Could we make this change in `master` first so it gets more CI exposure?

File: x-pack/plugin/autoscaling/src/internalClusterTest/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageIT.java [hunk 2]
```diff
@@ -178,7 +178,7 @@ public void testScaleFromEmptyLegacy() {
                     .put(INDEX_STORE_STATS_REFRESH_INTERVAL_SETTING.getKey(), "0ms")
                     .put(IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_SETTING.getKey() + "data_tier", "hot")
                     .build()
-            )
+            ).setWaitForActiveShards(ActiveShardCount.NONE)
```
Comment (idegtiarenko) [inline x-pack/plugin/autoscaling/src/internalClusterTest/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageIT.java:181]: This was waiting 30 seconds for a shards that could not be allocated. Following call updates settings for the correct allocation anyways...

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**